integration-tests/src/test/java/eu/tsystems/mms/tic/testframework/test/utils/AbstractSourceCodeTests.java

```diff
@@ -0,0 +1,50 @@
+package eu.tsystems.mms.tic.testframework.test.utils;
+
+import eu.tsystems.mms.tic.testframework.AbstractTestSitesTest;
+import eu.tsystems.mms.tic.testframework.common.PropertyManagerProvider;
+import eu.tsystems.mms.tic.testframework.core.pageobjects.testdata.PageWithNotExistingElement;
+import eu.tsystems.mms.tic.testframework.report.model.context.ErrorContext;
+import eu.tsystems.mms.tic.testframework.report.model.context.MethodContext;
+import eu.tsystems.mms.tic.testframework.report.model.context.ScriptSource;
+import eu.tsystems.mms.tic.testframework.testing.AssertProvider;
+import eu.tsystems.mms.tic.testframework.testing.PageFactoryProvider;
+import io.testerra.test.pretest_status.TestStatusTest;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Created on 2022-10-07
+ *
+ * @author mgn
+ */
+public class AbstractSourceCodeTests extends AbstractTestSitesTest implements TestStatusTest, AssertProvider, PropertyManagerProvider, PageFactoryProvider {
+
+    protected void doSimpleFail() {
+        ASSERT.fail("Simple failed");
+    }
+
+    protected void doPageFactoryFail() {
+        PAGE_FACTORY.createPage(PageWithNotExistingElement.class, this.getWebDriver());
+    }
+
+    protected void verifySourceCodeSnippet(String testMethodName, String fileName, String codeLine) {
+        List<ScriptSource> scriptSources = this.findMethodContexts(testMethodName)
+                .flatMap(MethodContext::readErrors)
+                .map(ErrorContext::getScriptSource)
+                .filter(Optional::isPresent)
```

Comment:
`flatMap(Optional::stream)` instead of `filter` + `map`?

Reply:
Ah yes, thanks for the hint.